sktime/forecasting/ets.py

```diff
@@ -0,0 +1,314 @@
+
+        Default is "aic".
+    allow_multiplicative_trend : bool, optional
+        If True, models with multiplicative trend are allowed when
+        searching for a model. Otherwise, the model space excludes them.
+        Default is False.
+    restrict : bool, optional
+        If True, the models with infinite variance will not be allowed.
+        Default is True.
+    additive_only : bool, optional
+        If True, will only consider additive models.
+        Default is False.
+    """
+
+    def __init__(
+        self,
+        error="add",
+        trend=None,
+        damped=False,
+        seasonal=None,
+        sp=None,
+        initialization_method="estimated",
+        initial_level=None,
+        initial_trend=None,
+        initial_seasonal=None,
+        bounds=None,
+        dates=None,
+        freq=None,
+        missing="none",
+        start_params=None,
+        maxiter=1000,
+        full_output=True,
+        disp=True,
+        callback=None,
+        return_params=False,
+        autofitting=False,
+        information_criterion='aic',
+        allow_multiplicative_trend=False,
+        restrict=True,
+        additive_only=False,
+        **kwargs
+    ):
+
+        # Model params
+        self.error = error
+        self.trend = trend
+        self.damped = damped
+        self.seasonal = seasonal
+        self.sp = sp
+        self.initialization_method = initialization_method
+        self.initial_level = initial_level
+        self.initial_trend = initial_trend
+        self.initial_seasonal = initial_seasonal
+        self.bounds = bounds
+        self.dates = dates
+        self.freq = freq
+        self.missing = missing
+
+        # Fit params
+        self.start_params = start_params
+        self.maxiter = maxiter
+        self.full_output = full_output
+        self.disp = disp
+        self.callback = callback
+        self.return_params = return_params
+        self.information_criterion = information_criterion
+        self.autofitting = autofitting
+        self.allow_multiplicative_trend = allow_multiplicative_trend
+        self.restrict = restrict
+        self.additive_only = additive_only
+
+        super(AutoETS, self).__init__()
+
+    def _fit_forecaster(self, y, X_train=None):
+
+        # Select model automatically
+        if self.autofitting:
+            # Initialise best forecaster parameters
+            best_forecaster = None
+            best_fitted_forecaster = None
+            best_information_criterion = np.inf
+
+            error_range = ['add', 'mul']
+            if self.allow_multiplicative_trend:
+                trend_range = ['add', 'mul', None]
+            else:
+                trend_range = ['add', None]
+            seasonal_range = ['add', 'mul', None]
+            damped_range = [True, False]
+
+            # Fit model, adapted from:
+            # https://github.com/robjhyndman/forecast/blob/master/R/ets.R
+            for error in error_range:
```

Comment:
This is probably something we want to parallelise. You could try to make this into a single for loop using [`itertools.product()`](https://docs.python.org/2/library/itertools.html#itertools.product) and joblib.

Reply:
```python
from itertools import product
from joblib import delayed, Parallel

def _iter(x, y):
    for a, b in product(x, y):
        if a == 1:
            continue
        yield a, b

def _fit(a, b):
    print(a, b)
    return a + b

x = [1, 2, 3]
y = [4, 5, 6]

results = Parallel(n_jobs=1)(delayed(_fit)(a, b) for a, b in _iter(x, y))
```